metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg_common.py

```diff
@@ -51,22 +43,34 @@ class IcebergProfilingConfig(ConfigModel):
     # include_field_sample_values: bool = True
 
 
+class IcebergSourceStatefulIngestionConfig(StatefulStaleMetadataRemovalConfig):
+    """Iceberg custom stateful ingestion config definition(overrides _entity_types of StatefulStaleMetadataRemovalConfig)."""
+
+    _entity_types: List[str] = pydantic.Field(default=["table"])
+
+
+class IcebergCatalogConfig(ConfigModel):
+    """
+    Iceberg catalog config.
+
+    https://py.iceberg.apache.org/configuration/
+    """
+
+    name: str = Field(
+        description="Name of catalog",
+    )
+    conf: Dict[str, str] = Field(
```

Comment:
AH!  I thought the type was always infered by the `uri`.  My bad since I've always been using REST, but I see what you mean when I look at Glue and DynamoDB. 👍🏻